### PR TITLE
chore: limit number of top level operations in services list

### PR DESCRIPTION
**Pull request by srikanthccv**
### Summary

Part of https://github.com/SigNoz/signoz/issues/4654

This is the schema of the `top_level_operations`

```sql
CREATE TABLE signoz_traces.top_level_operations
(
    `name` LowCardinality(String) CODEC(ZSTD(1)),
    `serviceName` LowCardinality(String) CODEC(ZSTD(1)),
    `time` DateTime DEFAULT now() CODEC(ZSTD(1))
)
ENGINE = ReplacingMergeTree
ORDER BY (serviceName, name)
TTL time + toIntervalMonth(1)
SETTINGS index_granularity = 8192
```

The table contains the most recent instance of the tuple (serviceName, name). And the time value is `now()`. We can use the `time` to discard the old operations that were created before the start time. 

This change

- Discard the old operation
- Use a maximum of 1500 operations for the `IN` query (this should be enough?) 
- Send all service ops to the client so we can show them to the user.

### CODE REVIEW

**Comment by srikanthccv** [comment]
We considered showing the users some of the selected top-level operations along with the warning text

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 1]
```diff
@@ -810,6 +824,22 @@ func (r *ClickHouseReader) GetServices(ctx context.Context, queryParams *model.G
 			defer func() { <-sem }()
 			var serviceItem model.ServiceItem
 			var numErrors uint64
+
+			// Even if the total number of operations within the time range is less and the all
+			// the top level operations are high, we want to warn to let user know the issue
+			// with the instrumentation
+			serviceItem.DataWarning = model.DataWarning{
+				TopLevelOps: (*allTopLevelOps)[svc],
```
Comment (ankitnayan) [inline pkg/query-service/app/clickhouseReader/reader.go:832]: sending all the values of `allTopLevelOps` in field `DataWarning`? This is a lot of unwanted data?

Comment (srikanthccv) [reply pkg/query-service/app/clickhouseReader/reader.go:832]: It's not unwanted. It is possible that In the last 30 minutes, the unique top-level ops may be well within the limit, however, overall customers sending operations that are cardinal enough to cause issues. We will show warning message even if the API works without an issue for the selected time but has underlying data problems.

Comment (ankitnayan) [reply pkg/query-service/app/clickhouseReader/reader.go:832]: >but has underlying data problems
where is this logic?

**Pull request merged**